scripts/prepare_sector_network.py

```diff
@@ -2446,7 +2459,10 @@ def add_rail_transport(n, costs):
     )
 
     # Define spatial for biomass and co2. They require the same spatial definition
-    define_spatial(pop_layout.index)
+    spatial = define_spatial(pop_layout.index, options)
+
+    if snakemake.config["foresight"] in ["myopic", "perfect"]:

```

Comment:
Planning for perfect optimisation, I like the optimism ;)

Reply:
ahahah